test/libsolidity/syntaxTests/specialFunctions/abidecode/abi_decode_array_of_tuples.sol

```diff
@@ -0,0 +1,15 @@
+contract C {
+    function f() pure public {
+        abi.decode("", ((uint)[2]));
```

Comment:
So, this is an array of tuples, namely of `(uint)` tuple, which is fine, but then nested tuples are not, e.g. `abi.decode(someBytes, ((uint)))`? But isn't `(uint)[]` equivalent to just `uint[]`? Also, would
```
abi.decode(someBytes, ((uint, uint)[]))
```
also be allowed?

Reply:
That results in another error: 
```
TypeError 2614: Indexed expression has to be a type, mapping or array (is tuple(type(uint256),type(uint256)))
```

Reply:
If index access is treated, specially, we should cover it with extra cases, e.g.:
- `(x = true ? uint : uint)[0]`
- `((x = true ? uint : uint))[0]`

Also this, which is disallowed, but IIRC we had a feature request to make `max`/`min` actually constant so it could become allowed in the future:
- `uint[type(x = true ? uint8 : uint8).max]`

Finally, since parentheses are special in some contexts, I'd also cover this (which should already be disallowed, but just in case):
- `((x = true ? uint : uint))`

Reply:
Added those cases.